File: app/src/main/java/com/yourname/gesturemusic/gesture/GestureArmingManager.kt
package com.yourname.gesturemusic.gesture

/**
 * Prevents accidental media commands by requiring a deliberate activation gesture.
 * Activation is intentionally separate from normal playback gestures.
 */
class GestureArmingManager(
    private val activeTimeoutMs: Long = 15_000L
) {
    var isArmed: Boolean = false
        private set

    private var armedAt: Long = 0L

    fun activate(now: Long): Boolean {
        isArmed = true
        armedAt = now
        return true
    }

    fun deactivate() {
        isArmed = false
        armedAt = 0L
    }

    fun update(now: Long): Boolean {
        if (isArmed && activeTimeoutMs > 0 && now - armedAt >= activeTimeoutMs) {
            deactivate()
        }
        return isArmed
    }

    fun touch(now: Long) {
        if (isArmed) armedAt = now
    }
}

File: src/gesture/GestureArmingManager.ts
/**
 * Prevents accidental media commands by requiring a deliberate activation gesture.
 * Ported directly from GestureArmingManager.kt
 */
export class GestureArmingManager {
  private activeTimeoutMs: number;
  public isArmed = false;
  private armedAt = 0;

  constructor(activeTimeoutMs = 15000) {
    this.activeTimeoutMs = activeTimeoutMs;
  }

  public activate(now: number): boolean {
    this.isArmed = true;
    this.armedAt = now;
    return true;
  }

  public deactivate() {
    this.isArmed = false;
    this.armedAt = 0;
  }

  public update(now: number): boolean {
    if (this.isArmed && this.activeTimeoutMs > 0 && now - this.armedAt >= this.activeTimeoutMs) {
      this.deactivate();
    }
    return this.isArmed;
  }

  public touch(now: number) {
    if (this.isArmed) {
      this.armedAt = now;
    }
  }
}
